### ISSUE TITLE
Doppler correction?

### ISSUE THREAD

**Issue by doctorvenables** [dbdexter-dev/meteor_demod#2]
Hi,
Love the project - finally managed to get workable .s files that my Rasp Pi can decode. It now sits in a batch job quite happily producing images.

I was wondering - there's another project produced by @cubehub called doppler, in which the original IQ is 'doppler corrected' which might mean that your demod program gains a lock faster. 

I've tried piping the output from rtl_fm and rtl_sdr to doppler and then to demod but the constellation diagram gets screwed up. 
With rtl_fm, I just get one quadrant with a circular patch in it, which moves from quadrant to quadrant.
With rtl_sdr, I get some trippy stuff...

Just a thought - maybe you two could talk to each other?
Thanks,
Mark

**Comment by dbdexter-dev** [comment]
Hi,
I've checked out `doppler`, and although it looks like doppler correction should help acquiring a lock faster, it also messes with the phase of the signal, so that makes keeping the carrier locked harder.
I tried using `doppler`, and while I do sometimes see the usual four patches on the constellation diagram, when the satellite is near its peak they all collapse into a single dot moving from quadrant to quadrant (as you noted). I've tried using other tools for QPSK demodulation, and they all show this exact behavior, so my guess is it's an unavoidable side effect of correcting for doppler. I'll look into it further, but unfortunately I don't think that'll ever become compatible with my tool. Thanks for the suggestion though!
